### PR TITLE
Timelock consensus

### PR DESCRIPTION
**Pull request by sudiksha27**
**Goals (and why)**:
TimeLock migration requires consensus to be achieved on all namespaces after a node is removed. This PR follows https://github.com/palantir/atlasdb/pull/4728, adds an endpoint on TimeLock to achieve consensus on a given set of namespaces.

**Implementation Description (bullets)**:

- For each namespace, fast forward timestamp to fresh timestamp for the namespace + 1 million

**Testing (What was existing testing like?  What have you done to improve it?)**:
Tested on IL stack.

**Concerns (what feedback would you like?)**:

**Where should we start reviewing?**:
NamespacedConsensus.java 

**Priority (whenever / two weeks / yesterday)**:
End of this week

### CODE REVIEW

File: timelock-impl/src/main/java/com/palantir/atlasdb/timelock/management/NamespacedConsensus.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import com.palantir.atlasdb.timelock.TimeLockServices;
+import com.palantir.atlasdb.timelock.TimelockNamespaces;
+
+public class NamespacedConsensus {
+
+    public static void achieveConsensusForNamespace(TimelockNamespaces timelockNamespaces,
+            String namespace) {
+        TimeLockServices timeLockServices = timelockNamespaces.get(namespace);
+        Long timestamp = timeLockServices.getTimelockService().getFreshTimestamp() + 1000000L;
+        timeLockServices.getTimestampManagementService().fastForwardTimestamp(timestamp);
```
Comment (sudiksha27) [inline timelock-impl/src/main/java/com/palantir/atlasdb/timelock/management/NamespacedConsensus.java:28]: add javadoc

File: timelock-impl/src/main/java/com/palantir/atlasdb/timelock/management/NamespacedConsensus.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    /**
+     * Gets a fresh timestamp for namespace from {@link TimestampService#getFreshTimestamp()} and fast forwards
+     * the timestamp by 1 million for the given namespace.
+     *
+     * @param namespace namespace for which consensus has to be achieved
+     */
+    public static void achieveConsensusForNamespace(TimelockNamespaces timelockNamespaces,
+            String namespace) {
+        TimeLockServices timeLockServices = timelockNamespaces.get(namespace);
+        Long timestamp = timeLockServices.getTimelockService().getFreshTimestamp() + 1000000L;
```
Comment (jeremyk-91) [inline timelock-impl/src/main/java/com/palantir/atlasdb/timelock/management/NamespacedConsensus.java:33]: nit: is there a need to explicitly box `timestamp`?

File: timelock-api/src/main/conjure/timelock-management-api.yml [hunk 3]
```diff
@@ -8,3 +8,8 @@ services:
       getNamespaces:
         http: POST /getNamespaces
         returns: set<string>
+
+      achieveConsensus:
+        http: POST /achieveConsensus
+        args:
+          namespaces: set<string>
```
Comment (jeremyk-91) [inline timelock-api/src/main/conjure/timelock-management-api.yml:15]: Might be useful to add Conjure docs here around how this operation handles leadership and/or failures.

File: timelock-api/src/main/conjure/timelock-management-api.yml [hunk 4]
```diff
@@ -8,3 +8,11 @@ services:
...
+        docs: |
+          The operation of achieving consensus on all namespaces is NOT atomic. In the case of failures, it can
+          be assumed that consensus was achieved on some namespaces.
```
Comment (jeremyk-91) [inline timelock-api/src/main/conjure/timelock-management-api.yml:18]:
```suggestion
          be assumed that consensus was achieved on some namespaces.

```

**Pull request merged**